tests/test_client.py

```diff
@@ -12,6 +12,25 @@
     assert response.text == "Hello, world!"
 
 
+@pytest.mark.asyncio
+async def test_get_base_url(server):
+    base_url = "http://127.0.0.1:8000/hello/"
+    path = "hummans/"
+    async with httpcore.Client(base_url=base_url) as client:
+        response = await client.get(path)
+    assert response.status_code == 200
+    assert response.text == "Hello, world!"
+    assert str(response.request.url) == "http://127.0.0.1:8000/hello/hummans/"
```

Comment:
Let's just have a single example in the test case, and use `http://127.0.0.1:8000/` as the base URL, and just do `.get('/hello')` or something like that.

(Also just use `str(response.url)` instead of `str(response.request.url)`)